extensions/amp-ad-network-adsense-impl/0.1/amp-ad-network-adsense-impl.js

```diff
@@ -75,17 +87,20 @@ export class AmpAdNetworkAdsenseImpl extends AmpA4A {
   getAdUrl() {
     const startTime = Date.now();
     const global = this.win;
+    const adClientId = this.element.getAttribute('data-ad-client');
     const slotId = this.element.getAttribute('data-amp-slot-index');
     const slotIdNumber = Number(slotId);
     const correlator = getCorrelator(this.win, slotId);
     const screen = global.screen;
     const slotRect = this.getIntersectionElementLayoutBox();
-    const visibilityState = viewerForDoc(this.getAmpDoc()).getVisibilityState();
+    const visibilityState = viewerForDoc(this.getAmpDoc())
+        .getVisibilityState();
     const adTestOn = this.element.getAttribute('data-adtest') ||
         isInManualExperiment(this.element);
     const format = `${slotRect.width}x${slotRect.height}`;
```

Comment:
Occurred to me that we need to handle the unlayoutCallback case so we can ensure that a subsequenty ad request for this slot does not include the slot's previous format.  Easiest way to accomplish this is to have unlayoutCallback remove its format from the array.  I believe we will still want an array (as opposed to say an object which would ease finding and removing a given format) as the expectation is likely that the order of the formats implies order of slots on the page.